linalg/ode.hpp

```diff
@@ -12,13 +12,91 @@
 #ifndef MFEM_ODE
 #define MFEM_ODE
 
+#include "../general/communication.hpp"
 #include "../config/config.hpp"
 #include "operator.hpp"
-#include "../general/communication.hpp"
+#include <vector>
+#include <memory>
 
 namespace mfem
 {
 
+/// An interface for storing state of previous timesteps
+class ODEStateData
+{
+public:
+   /// Get the maximum number of stored stages
+   virtual int MaxSize() const = 0;
+
+   /// Get the current number of stored stages
+   virtual int Size() const = 0;
+
+   /// Get the ith state vector
+   virtual void Get(int i, Vector &state) const = 0;
+
+   /// Get the ith state vector
+   virtual const Vector &Get(int i) const = 0;
+
+   /// Set the ith state vector
+   virtual void Set(int i, Vector &state) = 0;
+
+   /// Add state vector and increment state size
+   virtual void Add(Vector &state) = 0;
```

Comment:
We can probably avoid the copy in this method as well if we change its signature to simply return a reference to the new state vector. With this new interface, I expect `Add()` will need to be called earlier than now, however, that's probably a small change in existing code.

Reply:
I am not clear what changes are suggested.
Note that `Add` also increments a counter in the derived class.

Reply:
I am a bit confused by the suggestion.
Perhaps the function name is not fully covering what is meant. Perhaps a better name would be `append`.

FYI: I changed the name to append.

As mentioned earlier the cost of the copy is ignorable as this only happens at start-up.
Moreover, the use case in the unit test on line 170
`ode_solver->GetState().Append(uh[s]);`
would not be able to benefit form the suggested improvement.

My apologies for my reluctance, but seem to be out of my depth.
I do not have strong feeling regarding this, but I probably simply do not understand what is intended.
To avoid confusion, perhaps it would be best to make a concrete suggestion as was done below with the typo.